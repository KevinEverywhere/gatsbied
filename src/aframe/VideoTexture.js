import AFRAME from 'aframe'

AFRAME.registerComponent("video-texture", {
  init: function() {
    this.el.addEventListener("click", (e) => {
      console.log(e);
      console.log(this.el.getAttribute('src'))
      // if(document.querySelector(this.el.getAttribute('src')).paused){
        document.querySelector(this.el.getAttribute('src')).play()
      // }else{
      //   document.querySelector(this.el.getAttribute('src')).pause()
      // }
      // this.el.setAttribute("scale", scale)
    })
  }
})
