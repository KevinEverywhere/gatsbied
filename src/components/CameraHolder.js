import React from 'react'

class CameraHolder extends React.PureComponent {
  constructor(props){
    super();
    this.state={activeCamera:null}
  }
  stopVideo=(videoElem) =>{
    let stream = videoElem.srcObject;
    let tracks = stream.getTracks();
    tracks.forEach(function(track) {
      track.stop();
    });
    videoElem.srcObject = null;
  }
  makeActive=()=>{
    const me=this;
    navigator.mediaDevices.getUserMedia({ audio: true, video: { facingMode: this.props.facingMode }})
    .then(function(stream) {
      var video = document.getElementById(me.props.camera);
      if ("srcObject" in video) {
        video.srcObject=stream;
      }else{
        video.src = window.URL.createObjectURL(stream);
      }
      video.onloadedmetadata = function(e) {
        video.play();
      };
    })
    .catch((oops)=>{
      console.log('bad camera')
    });
  }
  makeInactive=()=>{
    try{
      var video = document.getElementById(this.props.camera);
      this.stopVideo(video);
    }catch(oops){
       // video not playing
    }
  }
  setVideo=()=>{
    if(navigator.mediaDevices && navigator.mediaDevices.getUserMedia){
      const srcTarget = `side:double;src: #${this.props.camera}`
      this.setState({
        activeCamera:(
          <a-entity id={this.props.id} position={this.props.position} rotation={this.props.rotation}>
            <a-entity position="0 2 0"
            geometry="primitive: plane; width: 240; height:180"
            material={srcTarget}></a-entity>
          </a-entity>
        )
      })
      this.makeActive();
      if(this.props.active!==true){
        this.makeInactive();
      }
    }
  }
  componentWillUnmount(){
    this.makeInactive();
  }
  componentDidMount(){
    import('aframe')
      .then(() => {
        import('aframe-extras')
          .then(() => {
            this.setVideo();
          })
          .catch((oops)=>{
            console.log('bad camera')
          });
      })
      .catch((error) => console.error(error));
    }
  render() {
    return this.state.activeCamera
  }
}

export default CameraHolder

// <div class="inputdialog">
//   <input class="biginput" type="file" accept="video/*" capture="camera">
// </div>
