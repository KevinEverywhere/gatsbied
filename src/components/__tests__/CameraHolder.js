import React from "react"
import renderer from "react-test-renderer"
import CameraHolder from "../CameraHolder"

describe("CameraHolder", () => {
  it("renders correctly", () => {
    Promise.all([
      import('aframe'),
      import('aframe-extras')
     ]).then({
    const tree = renderer
      .create(<CameraHolder
        active="true"
        facingMode="environment"
        position="300 20 -300"
        rotation="0 315 0"
        camera="webcam"
        id="cameraRig"
        />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
//
// Promise.all([
//   import('aframe'),
//   import('aframe-extras')
//  ]).then(([
//    aframe,
//    aframeExtras
//   ]) => {
//    console.log(Import1);
//    /* CODE HERE*/
//   });
