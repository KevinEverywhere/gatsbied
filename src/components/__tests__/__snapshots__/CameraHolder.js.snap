// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CameraHolder renders correctly 1`] = `null`;
